src/core/js/models/adaptModel.js

```diff
@@ -112,57 +112,45 @@ define([
-                    }));
-                } else {
-                    isComplete = (_.filter(children, function(child) {
-                        return !child.get('_isComplete') && !child.get('_isOptional');
-                    }).length >= requireCompletionOf);
-                }
-
-                this.set({_isComplete: isComplete});
-
-                Adapt.checkedCompletion();
-            }, this));
+            _.defer(_.bind(this.checkCompletionStatusFor, this, "_isComplete"));
         },
 
         checkInteractionCompletionStatus: function () {
-            //defer to allow other change:_isInteractionComplete handlers to fire before cascasing to parent
+            //defer to allow other change:_isInteractionComplete handlers to fire before cascading to parent
             Adapt.checkingCompletion();
-            _.defer(_.bind(function() {
-                var isInteractionComplete = false;
-                var children = this.getAvailableChildModels();
-                //number of mandatory children that must be complete or -1 for all
-                var requireCompletionOf = this.get("_requireCompletionOf");
-
-                if (requireCompletionOf === -1) {
-                    // Check if any return _isInteractionComplete:false
-                    // If not - set this model to _isInteractionComplete: true
-                    isInteractionComplete = (_.find(children, function(child) {
-                        return child.get('_isInteractionComplete') === false && child.get('_isOptional') === false;
-                    }) === undefined);
-                } else {
-                    isInteractionComplete = (_.filter(children, function(child) {
-                        return child.get('_isInteractionComplete') === true && child.get('_isOptional') === false;
-                    }).length >= requireCompletionOf);
-                }
+            _.defer(_.bind(this.checkCompletionStatusFor, this, "_isInteractionComplete"));
+        },
 
-                this.set({_isInteractionComplete:isInteractionComplete});
-                Adapt.checkedCompletion();
+        /**
+         * Function for checking whether the supplied completion attribute should be set to true or false. 
+         * It iterates over our immediate children, checking the same completion attribute on any mandatory child
+         * to see if enough/all of them them have been completed. If enough/all have, we set our attribute to true; 
+         * if not, we set it to false.
+         * @param {string} [completionAttribute] Either "_isComplete" or "_isInteractionComplete". Defaults to "_isComplete" if not supplied.
+         */        
+        checkCompletionStatusFor: function(completionAttribute) {
+            // in the absence of support for default function parameters in IE & Safari v9...
+            if (completionAttribute === void 0) completionAttribute = "_isComplete";
+
+            var complete = false;
```

Comment:
(Minor) It might be slightly more readable to call this variable `isComplete`.

Reply:
The only reason I changed this is that one of the attributes it's checked is `_isComplete` so I felt it was a bit too close to that...

Reply:
I've gone for `completed` instead which seems slightly clearer?